synapse/models/infotech.py

```diff
@@ -60,6 +60,14 @@
 cpe22_regex = regex.compile(f'cpe:/{COMPONENT_LIST}', regex.VERBOSE | regex.IGNORECASE)
 cpe23_regex = regex.compile(s_scrape._cpe23_regex, regex.VERBOSE | regex.IGNORECASE)
 
+def isValidCpe22(text):
+    rgx = cpe22_regex.match(text)
+    return rgx is not None and rgx.group() == text
+
+def isValidCpe23(text):
+    rgx = cpe23_regex.match(text)
+    return rgx is not None and rgx.group() == text
```

Comment:
Should be able to use `fullmatch` rather than `match` here and skip the extra checks